nilearn/decomposition/tests/conftest.py

```diff
@@ -0,0 +1,319 @@
+"""Fixtures for decomposition tests."""
+
+from typing import Union
+
+import numpy as np
+import pytest
+from nibabel import Nifti1Image
+
+from nilearn.maskers import MultiNiftiMasker, SurfaceMasker
+from nilearn.surface import PolyMesh, SurfaceImage
+from nilearn.surface.tests.test_surface import flat_mesh
+
+SHAPE_SURF = {"left": (15, 5), "right": (10, 4)}
+RANDOM_STATE = 42
+N_SUBJECTS = 3
+# TODO
+# some fixtures or tests start breaking if some of those values below
+# are changed
+N_SAMPLES = 5
+N_COMPONENTS = 4
+
+
+def _decomposition_mesh() -> PolyMesh:
+    """Return a mesh to use for decomposition tests."""
+    return PolyMesh(
+        left=flat_mesh(*SHAPE_SURF["left"]),
+        right=flat_mesh(*SHAPE_SURF["right"]),
+    )
+
+
+@pytest.fixture
+def decomposition_mesh() -> PolyMesh:
+    """Return a mesh to use for decomposition tests."""
+    return _decomposition_mesh()
+
+
+@pytest.fixture
+def decomposition_mask_img(
+    data_type: str,
+    decomposition_mesh: PolyMesh,
+    affine_eye: np.ndarray,
+    shape_3d_large,
+) -> Union[SurfaceImage, Nifti1Image]:
+    """Return a mask for decomposition."""
+    if data_type == "surface":
+        mask_data = {
+            "left": np.ones(
+                (decomposition_mesh.parts["left"].coordinates.shape[0],)
+            ),
+            "right": np.ones(
+                (decomposition_mesh.parts["right"].coordinates.shape[0],)
+            ),
+        }
+        return SurfaceImage(mesh=decomposition_mesh, data=mask_data)
+
+    # TODO
+    # setting the shape of the mask to be a bit different
+    # shape_3d_large that is used for the data
+    # to force resampling
```

Comment:
You mean for images ?

Reply:
yeah all of those fixtures we use for testing decomposition on volume data, tend to generate data (image and mask) that have same the same shape and affine, so we never properly test that things go as expected if the we had a mask with a different field of view from the data (or the other way around)

I am pretty sure that we do not see it in terms of coverage because this is handled by the maskers and so this part of the code base is tested in other ways, but I'd prefer our tests for decomposition to not just test the "happy path".